archive/extract.go

```diff
@@ -10,29 +10,23 @@ import (
 	"github.com/pkg/errors"
 )
 
-var systemUmask int
-
-func init() {
-	// Avoid umask from changing the file permissions in the tar file.
-	systemUmask = setUmask(0)
-	_ = setUmask(systemUmask)
-}
-
 type PathMode struct {
 	Path string
 	Mode os.FileMode
 }
 
-// Extract reads all entries from TarReader and extracts them to the filesystem
-func Extract(tr TarReader) error {
+// Extract reads all entries from TarReader and extracts them to the filesystem.
+// The umask must be unset before calling this function on unix, to ensure that files have the correct file mode.
```

Comment:
Do we want to add that we can call `SetUmask` after this function to return the original value?

Reply:
```suggestion
// The umask must be unset before calling this function on unix, to ensure that files have the correct file mode. SetUmask can be used to set and unset the umask.
```